mvc/views/tag-helpers/intro.rst

```diff
@@ -1,8 +1,281 @@
+
+As soon as a Tag Helper attribute is entered, the tag and attribute fonts change. Using the default Visual Studio "Blue" or "Light" color theme, the font is bold purple. If you're using the "Dark" theme the font is bold teal. The images in this document were taken using the default theme.
+
+.. image:: intro/_static/labelaspfor.png
+
+You can enter the Visual Studio *CompleteWord* shortcut (Ctrl +spacebar is the `default <https://msdn.microsoft.com/en-us/library/da5kh0wa.aspx>`__) inside the double quotes (""), and you are now in C#, just like you would be in a C# class. IntelliSense displays all the methods and properties on the page model. In the image below, I'm editing the ``Register`` view, so the ``RegisterViewModel`` is available.
+
+.. image:: intro/_static/intellemail.png
+
+IntelliSense lists the properties and methods available to the model on the page. The rich IntelliSense environment helps you select the CSS class:
+
+.. image:: intro/_static/iclass.png
+
+.. image:: intro/_static/intel3.png
+
+Tag Helpers compared to HTML Helpers
+---------------------------------------------
+
+Tag Helpers attach to HTML elements in Razor views, while `HTML Helpers <http://stephenwalther.com/archive/2009/03/03/chapter-6-understanding-html-helpers>`__ are invoked as methods interspersed with HTML in Razor views. Consider the following Razor markup, which creates an HTML label with the CSS class "caption":
+
+.. code-block:: html
+
+	@Html.Label("FirstName", "First Name:", new {@class="caption"})
+
+The at (``@``) symbol tells Razor this is the start of code. The next two parameters ("FirstName" and "First Name:") are strings, so `IntelliSense <https://msdn.microsoft.com/en-us/library/hcw1s69b.aspx>`_ can't help. The last argument:
+
+.. code-block:: html
+
+  new {@class="caption"}
+  
+Is an anonymous object used to represent attributes. Because **class** is a reserved keyword in C#, you use the ``@`` symbol to force C# to interpret "@class=" as a symbol (property name). To a front-end designer (someone familiar with HTML/CSS/JavaScript and other client technologies but not familiar with C# and Razor), most of the line is foreign. The entire line must be authored with no help from IntelliSense.
+  
+Using the ``LabelTagHelper``, the same markup can be written as:
+
+.. code-block:: html
+
+	<label class="caption" asp-for="FirstName"></label>
+  
+With the Tag Helper version, as soon as you enter ``<l`` in the Visual Studio editor, IntelliSense displays matching elements:
+
+.. image:: intro/_static/label.png 
+
+
+IntelliSense helps you write the entire line. The ``LabelTagHelper`` also defaults to setting the content of the ``asp-for`` attribute value ("FirstName") to "First Name"; It converts camel-cased properties to a sentence composed of the property name with a space where each new upper-case letter occurs. In the following markup:
+
+ .. code-block:: html
+
+	<label class="caption" asp-for="FirstName"></label>
```

Comment:
Hmmmm might be worth having an image. Showcasing the `TagHelper` syntax without the coloring isn't as epic. :smile: